platform/util/platform-util/src/main/java/org/codice/ddf/platform/util/TemporaryFileBackedOutputStream.java

```diff
@@ -83,12 +101,30 @@ public void close() throws IOException {
         }
         try {
             fileBackedOutputStream.close();
-            fileBackedOutputStream.reset();
+            reset();
         } finally {
             isClosed = true;
         }
     }
 
+    /**
+     * Reset fileBackedOutputStream and retry if it fails.
+     */
+    @SuppressWarnings("unchecked")
+    private void reset() {
+
+        RetryPolicy retryPolicy = new RetryPolicy().retryOn(IOException.class)
+                .withBackoff(INITIAL_RETRY_SLEEP, MAX_DELAY, INITIAL_RETRY_SLEEP_UNIT)
+                .withMaxRetries(MAX_RETRY_ATTEMPTS);
+
+        Failsafe.with(retryPolicy)
+                .onFailedAttempt(throwable -> LOGGER.debug("failed to reset FBOS, will retry",
+                        throwable))
+                .onFailure(throwable -> LOGGER.debug("failed to reset FBOS", throwable))
```

Comment:
There is no public method to get the underlying temp file. The only way would be to use reflection to break encapsulation. However, the underlying FileBackedOutputStream does include the filename in an exception. Was there other information that you want include in the onFailure log statement? There really isn't any other info that I could add.